core/templates/pages/contributor-dashboard-page/modal-templates/certificate-download-modal.component.ts

```diff
@@ -0,0 +1,329 @@
+    this.activeModal.close();
+  }
+
+  downloadCertificate(): void {
+    this.errorsFound = false;
+    this.errorMessage = '';
+    if (
+      new Date(this.fromDate) >= new Date(this.toDate) ||
+      !this.fromDate ||
+      !this.toDate
+    ) {
+      this.errorsFound = true;
+      this.errorMessage = 'Invalid date range.';
+      return;
+    }
+    if (new Date() < new Date(this.toDate)) {
+      this.errorsFound = true;
+      this.errorMessage = 'Please select a \'To\' date that is earlier than ' +
+        'today\'s date';
+      return;
+    }
+    this.certificateDownloading = true;
+    this.contributionAndReviewService.downloadContributorCertificateAsync(
+      this.username,
+      this.suggestionType,
+      this.languageCode,
+      this.fromDate,
+      this.toDate
+    ).then((response: ContributorCertificateResponse) => {
+      this.createCertificate(response);
+      this.certificateDownloading = false;
+    }).catch(() => {
+      this.errorsFound = true;
+      this.certificateDownloading = false;
+      this.errorMessage = (
+        'Not able to download contributor certificate');
+    });
+  }
+
+  disableDownloadButton(): boolean {
+    return this.fromDate === undefined || this.toDate === undefined;
+  }
+
+  createCertificate(response: ContributorCertificateResponse): void {
+    const canvas = document.createElement('canvas');
+    const currentDate = new Date();
+    const dateOptions: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric'
+    };
+    // Textual parts are starting when y coordinate is equals to 350.
+    let linePosition = 350;
+    const image = new Image(this.LOGO_WIDTH, this.LOGO_HEIGHT);
+    image.src = '/assets/images/contributor_dashboard/oppia-logo.jpg';
+    canvas.width = this.CERTIFICATE_WIDTH;
+    canvas.height = this.CERTIFICATE_HEIGHT;
+    const ctx = canvas.getContext('2d');
+
+    if (ctx === null) {
+      throw new Error('Canvas context not found.');
+    }
+
+    image.onload = () => {
+      ctx.fillStyle = '#D0E1F0';
+      ctx.fillRect(
+        this.CANVAS_ROOT_COORDINATE,
+        this.CANVAS_ROOT_COORDINATE,
+        this.CERTIFICATE_WIDTH,
+        this.CERTIFICATE_HEIGHT
+      );
+
+      ctx.fillStyle = '#F7FDFF';
+      ctx.fillRect(
+        this.INNER_BOX_COORDINATE,
+        this.INNER_BOX_COORDINATE,
+        this.CERTIFICATE_INNER_BOX_WIDTH,
+        this.CERTIFICATE_INNER_BOX_HEIGHT
+      );
+
+      ctx.drawImage(
+        image,
+        this.LOGO_X_COORDINATE,
+        this.LOGO_Y_COORDINATE,
+        this.LOGO_WIDTH,
+        this.LOGO_HEIGHT
+      );
+
+      ctx.font = '37px Capriola';
+      ctx.fillStyle = '#00645C';
+      ctx.textAlign = 'center';
+      ctx.fillText(
+        'CERTIFICATE OF APPRECIATION',
+        this.CERTIFICATE_MID_POINT,
+        linePosition
+      );
+
+      ctx.font = '30px Capriola';
+      ctx.fillStyle = '#8F9899';
+      // Increase y coordinate by 150.
```

Comment:
Comments like this seem a bit redundant. Are they needed?

Reply:
Removed